sheets/1/exercise-3.tex

```diff
@@ -39,40 +39,47 @@
-                  \item Klassen sind in Java quasi die Nomen und sollten Groß geschreiben werden. Alles andere Klein.
-                  %TODO nomen rausnehmen ?
-                  \item Namen sollten aus ganzen Wörtern bestehen. Ausgenommen es ist eine sehr übliche bezeichnung wie \texttt{URL} oder \texttt{HTML}.\\
-                  Außerdem wird bei manchen Variablen einzelne buchstaben wie \texttt{i,j,k,m,n,c,d,e} verwendet. Solche stellen kommen Später.
-                  %TODO: sollte das drin sein?
-                  \item Operationen sollten Verben sein.
+                  \item Klassen beschreiben Objekte und ihre Namen sollten mit einem Großbuchstaben anfangen. Andere Namen, wie die von Variablen oder Operationen sollten mit einem klein Buchstaben anfangen.
+                  \item Namen sollten aus ganzen Wörtern bestehen.
+                  \item Operationen sollten Verben sein. Da sie dinge enthalten die ein Objekt tun kann.
                   \item Queries beginnen häufig mit \texttt{get}
                   \item Kommandos beginnen häufig mit \texttt{set}, wenn sie dazu dienen direkt einen Wert zu setzen.
               \end{itemize}
           \end{Infobox}
     \item \textbf{[1]} Ein Befehl endet immer mit einem ; und muss in einem Codeblock stehen.
     \item \textbf{[2]} Das erzeugen eines Objekts wird mit dem Schlüsselwort new eingeführt. Darauffolgt der Name der Klasse und (). Manchmal werden in die () noch Argumente geschreiben, mit Dingen die das Objekt von Anfang an braucht.
           \begin{lstlisting}[title=\textbf{Konstruktor Syntax},firstnumber=5]
-        new <Typ Name>();
+    new <Typ Name>();
+    new <Typ Name>(<Wert>);
+    new <Typ Name>(<Wert>,<Wert>);
+    \end{lstlisting}
+    \begin{lstlisting}[title=\textbf{Beispiel (1/3) Konstruktor},firstnumber=2,frame=lr]
+        new PDFCreator("jvk title",pdfCollector);
     \end{lstlisting}
-    \item \textbf{[2]} Wenn wir ein Objekt länger als nur einmal brauchen müssen wir uns dieses Merken. Dazu verwenden wir eine Variable. Um eine zu erstellen. Schreiben wir zuerst die Klasse zu der das Objekt gehört. Gefolgt von dem frei wälbaren Namen den die Variable haben soll. Mit einem = lassen wir sie dann auf ein Objekt zeigen, indem wir entweder das new Objekt() direkt dahinter schreiben oder eine andere
+    \item \textbf{[2]} Wenn wir ein Objekt länger als nur einmal brauchen müssen wir uns dieses Merken. Dazu verwenden wir eine Variable. Um eine zu erstellen. Schreiben wir zuerst die Klasse zu der das Objekt gehört. Gefolgt von dem frei wälbaren Namen den die Variable haben soll. Mit einem = weisen wir ein Objekt einer Variable zu, indem wir entweder das new Objekt() direkt dahinter schreiben oder eine andere
           Variable die bereits einen Wert hat.
+          Wenn wir ein Objekt einem Wert zugewiesen haben, können wir es unter dem Namen der Variable wiederfinden.
           \begin{lstlisting}[title=\textbf{Variablen Syntax}]
-    <Typ>
+    <Typ> <Name>;
```

Comment:
Vielleicht in kommentaren hinter der Zeile Erklären was passiert.
// Hier Erstellen der Variable

Reply:
ich wollte kommentare da nochnicht verwenden, weil sie da noch nicht erklährt wurden. teoretisch könnten wir die erklährung nach vor ziehen.

Reply:
Das wäre dann vielleicht sinnvoll die Kommentare möglichst am Anfang zu erklären.